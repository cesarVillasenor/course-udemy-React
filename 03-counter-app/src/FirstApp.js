import React from 'react';  
import PropTypes from  'prop-types';

// import React, { Fragment } from 'react';  

// Functional Components
// props 
const FirstApp = ({ greeting, subtitle }) => {
    // const greeting = 'Hello World';




    return (
        <>
            <h1>{ greeting }  !!!</h1>
            {/* <pre> {  JSON.stringify( greeting, null, 3 ) } </pre> */}
            <p>{ subtitle }</p>
        </>
    );


} 

FirstApp.propTypes={
    greeting: PropTypes.string.isRequired
    // other = PropTypes.number.isRequired,
}

FirstApp.defaultProps = {
    subtitle: "I'm a Subtitle" ,
}

export default FirstApp;