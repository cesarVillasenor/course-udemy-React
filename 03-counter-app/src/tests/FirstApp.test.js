import React from 'react';
import { shallow } from 'enzyme';
// import { render } from '@testing-library/react';
import FirstApp from '../FirstApp';

describe('Test for <FirstApp />', () => {

    // test('Should return the messages "Hello i am Goku" ', () => {
    //     const greeting =  "Hello i am Goku";
    //     const { getByText } =  render( <FirstApp greeting={ greeting } />);
    //     expect( getByText( greeting ) ).toBeInTheDocument();    
    // });
   
    test('should show <FirstApp /> correctly', () => {

        const greeting = "Hello i am Goku";
        const wrapper = shallow(<FirstApp greeting={ greeting }/>);

        expect( wrapper ).toMatchSnapshot();

        
    });

    test('should  show subtitle sent by props', () => {
        const greeting = "Hello i am Goku";
        const subtitle = "I am subtitle";

        const wrapper = shallow(
            <FirstApp 
            greeting={ greeting }
            subtitle={ subtitle } 
            />
            );
        
        const pText =  wrapper.find('p').text();
        // console.log(pText);
        expect( pText ).toBe( subtitle )
        
    })
    
    

});