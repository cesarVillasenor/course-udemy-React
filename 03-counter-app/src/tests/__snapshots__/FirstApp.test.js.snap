// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <FirstApp /> should show <FirstApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hello i am Goku
      !!!
  </h1>
  <p>
    I'm a Subtitle
  </p>
</Fragment>
`;
